File: tictactoe.client/src/types/gameDto.ts
import { CharacterIcon, GameState, GameResult } from './game';

export interface BaseDto {
  id: number;
}

export interface GameDto extends BaseDto {
  gameId: string;
  player1Id: string;
  player1Name: string;
  player1Character: CharacterIcon;
  player2Id?: string;
  player2Name?: string;
  player2Character?: CharacterIcon;
  isPrivate: boolean;
  state: GameState;
  result: GameResult;
  winnerId?: string;
  winnerName?: string;
  createdAt: Date;
  startedAt?: Date;
  endedAt?: Date;
  totalMoves: number;
  duration?: number; // Duration in milliseconds (converted from TimeSpan)
}

export interface CreateGameRequest {
  player1Name: string;
  player1Character: CharacterIcon;
  isPrivate: boolean;
}

export interface UpdateGameRequest {
  player2Id?: string;
  player2Name?: string;
  player2Character?: CharacterIcon;
  state?: GameState;
  result?: GameResult;
  winnerId?: string;
  winnerName?: string;
  startedAt?: Date;
  endedAt?: Date;
  totalMoves?: number;
  duration?: number;
} 

File: TicTacToe.Server/DTOs/GameDto.cs
using TicTacToe.Server.Models;

namespace TicTacToe.Server.DTOs;

public class GameDto : BaseDto
{
    public string GameId { get; set; } = string.Empty;
    public string Player1Id { get; set; } = string.Empty;
    public string Player1Name { get; set; } = string.Empty;
    public CharacterIcon Player1Character { get; set; }
    public string? Player2Id { get; set; }
    public string? Player2Name { get; set; }
    public CharacterIcon? Player2Character { get; set; }
    public bool IsPrivate { get; set; }
    public GameState State { get; set; }
    public GameResult Result { get; set; }
    public string? WinnerId { get; set; }
    public string? WinnerName { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? EndedAt { get; set; }
    public int TotalMoves { get; set; }
    public TimeSpan? Duration { get; set; }
} 